osidb/tests/test_mixins.py

```diff
@@ -14,21 +16,177 @@
 pytestmark = pytest.mark.unit
 
 
+class TestACLAlertMixin:
+    """
+    negative tests of ACL validations
+    the positive tests are all over the test suite
+    """
+
+    def group2acl(self, group):
+        """
+        translate the human-readable LDAP group
+        name to the corresponding UUID ACL hash
+        """
+        return uuid.UUID(generate_acls([group])[0])
+
+    def create_flaw(self, acl_read=None, acl_write=None, save=True):
+        """
+        shortcut for creating a flaw with the given ACLs
+        """
+        kwargs = {}
+        if acl_read is not None:
+            kwargs["acl_read"] = [self.group2acl(group) for group in acl_read]
+        if acl_write is not None:
+            kwargs["acl_write"] = [self.group2acl(group) for group in acl_write]
+        return FlawFactory(**kwargs) if save else FlawFactory.build(**kwargs)
+
+    def test_empty_acl_read(self):
+        """
+        test that an empty read ACL is not allowed
+        """
+        with pytest.raises(ValidationError) as e:
+            FlawFactory(acl_read=[])
+        assert "'acl_read': ['This field cannot be blank.'" in str(e)
+
+    def test_empty_acl_write(self):
+        """
+        test that an empty write ACL is not allowed
+        """
+        with pytest.raises(ValidationError) as e:
+            FlawFactory(acl_write=[])
+        assert "'acl_write': ['This field cannot be blank.'" in str(e)
```

Comment:
nitpick: we should try to use `pytest.raises(..., match="foo")` instead of `"foo" in str(e)`

Reply:
Good suggestion. Will do.